python/cudf/cudf/core/column/column.py

```diff
@@ -1938,6 +1950,9 @@ def as_column(
         else:
             data = as_column(cupy.asarray(arbitrary), nan_as_null=nan_as_null)
 
+        if delayed_cast and dtype is not None:
```

Comment:
If I'm understanding correctly, `delayed_cast` should only be getting set to True if we enter the above check for `dtype is not None`, so this could be

```suggestion
        if delayed_cast:
```

Unless I'm missing something where the `np.dtype(dtype)` call failing would set `dtype` to None?

Reply:
Yup, you got it right.